### PR TITLE
Switch cross references to a ValueStore.

### PR DESCRIPTION
**Pull request by jonmeow**
As I get ready to add imports, I need an API to add values and this offers a consistent way of doing so.

### CODE REVIEW

File: toolchain/sem_ir/file.cpp [hunk 1]
```diff
@@ -44,10 +44,11 @@ auto TypeInfo::Print(llvm::raw_ostream& out) const -> void {
 File::File(SharedValueStores& value_stores)
     : value_stores_(&value_stores),
       filename_("<builtins>"),
-      // Builtins are always the first IR, even when self-referential.
-      cross_reference_irs_({this}),
       type_blocks_(allocator_),
       inst_blocks_(allocator_) {
+  CARBON_CHECK(cross_reference_irs_.Add(this) == CrossReferenceIRId::Builtins)
+      << "Builtins must be the first IR, even if self-referential";
```
Comment (zygoloid) [inline toolchain/sem_ir/file.cpp]: I would prefer that we don't put code with side-effects in a `CARBON_CHECK`: as a reader I expect that the `CHECK` expression is only computing the value that we're validating and will skip over it when looking for the behavior of the function.

File: toolchain/sem_ir/file.cpp [hunk 2]
```diff
@@ -72,13 +73,12 @@ File::File(SharedValueStores& value_stores, std::string filename,
...
       filename_(std::move(filename)),
-      // Builtins are always the first IR.
-      cross_reference_irs_({builtins}),
       type_blocks_(allocator_),
       inst_blocks_(allocator_) {
   CARBON_CHECK(builtins != nullptr);
-  CARBON_CHECK(builtins->cross_reference_irs_[0] == builtins)
-      << "Not called with builtins!";
+  CARBON_CHECK(cross_reference_irs_.Add(builtins) ==
+               CrossReferenceIRId::Builtins)
```
Comment (zygoloid) [inline toolchain/sem_ir/file.cpp]: Same comment here.

**Review by zygoloid — APPROVED** [review]
Tiny change then LG.

File: toolchain/sem_ir/file.cpp (hunk 1 above)
Comment (jonmeow) [reply toolchain/sem_ir/file.cpp]: Done

File: toolchain/sem_ir/file.cpp (hunk 2 above)
Comment (jonmeow) [reply toolchain/sem_ir/file.cpp]: Done

**Pull request merged**